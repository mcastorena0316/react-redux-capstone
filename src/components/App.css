.App {
  text-align: center;
  color:white
}
