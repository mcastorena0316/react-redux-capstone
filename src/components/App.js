import React from 'react';
import './App.css';
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from 'react-router-dom';
import MissionList from '../containers/MissionList';
import Mission from '../containers/Mission';
import Header from './Header';
import About from './About';

function App() {
  return (
    <Router>
      <div className="App">
        <Header />
      </div>
      <Switch>
        <Route path="/" exact component={MissionList} />
        <Route path="/about" exact component={About} />
        <Route path="/mission/:id" component={Mission} />
      </Switch>
    </Router>
  );
}

export default App;
