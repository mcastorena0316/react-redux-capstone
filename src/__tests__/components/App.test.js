import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
// import configureStore from 'redux-mock-store';
import App from '../../components/App';
import '../../setupTests';

describe('<App />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<App />);
      const component = wrapper.dive();
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
