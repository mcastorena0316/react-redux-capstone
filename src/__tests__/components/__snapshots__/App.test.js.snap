// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render() renders the component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div
    className="App"
  >
    <Header />
  </div>
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(MissionList)",
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/about"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Mission)",
          "type": [Function],
        }
      }
      path="/mission/:id"
    />
  </Switch>
</Router>
`;
